localstack/services/dynamodb/provider.py

```diff
@@ -309,6 +295,24 @@ def on_after_init(self):
             endpoint=self.handle_shell_ui_request,
         )
 
+    def _modify_ddblocal_arns(self, chain, context: RequestContext, response: Response):
+        """A service response handler that modifies the dynamodb backend response."""
+        if response_content := response.get_data(as_text=True):
+            # fix the table and latest stream ARNs (DynamoDBLocal hardcodes "ddblocal" as the region)
+            content_replaced = re.sub(
+                r'("TableArn"|"LatestStreamArn"|"StreamArn")\s*:\s*"arn:([a-z-]+):dynamodb:ddblocal:([^"]+)"',
+                rf'\1: "arn:\2:dynamodb:{aws_stack.get_region()}:\3"',
+                response_content,
+            )
+            if content_replaced != response_content:
+                response.data = content_replaced
+                context.service_response = (
+                    None  # make sure the service response is parsed again later
+                )
+
+        # update x-amz-crc32 header required by some clients
+        response.headers["x-amz-crc32"] = crc32(response.data) & 0xFFFFFFFF
```

Comment:
Since you are directly calculating the checksum here, the `fix_headers_for_updated_response` remains as dead code, right? And the same goes for `localstack.services.dynamodb.utils#calculate_crc32`, I think.

Reply:
yep good catch, will remove